dist/datasource.js

```diff
@@ -141,6 +141,7 @@ function (angular, _, dateMath, moment) {
       var dataSource = this;
       var from = dateToMoment(options.range.from, false);
       var to = dateToMoment(options.range.to, true);
+      dataSource.periodGranularity = (options.timezone!="" && options.timezone!="browser"?options.timezone:dataSource.periodGranularity);
```

Comment:
dataSource.periodGranularity = (
 if(dataSource.periodGranularity=="dashboard" )
{
 if(options.timezone!=""){
alert error - grafna not sending timezone
 }
 else if (options.timezone=='browser'){
 use js to figure out timezone
 }
 else{
 options.timezone
 }
}
else{
dataSource.periodGranularity
}